datafusion/proto/proto/datafusion.proto

```diff
@@ -545,7 +545,7 @@ message InListNode {
 
 enum ScalarFunction {
   Abs = 0;
-  Acos = 1;
+  // 1 was Acos
```

Comment:
Should we consider keeping it, perhaps by renaming it to Deprecated_Acos, and then decode it into the new ScalarUDF to maintain better backward compatibility?  

We might need to mark the `BuiltinScalarFunction::Acos` as deprecated as well.

Reply:
> hould we consider keeping it, perhaps by renaming it to Deprecated_Acos, and then decode it into the new ScalarUDF to maintain better backward compatibility?

I followed the way that how #9216 handed it, gentle ping on @alamb, maybe you have some thoughts on this

Reply:
👌

Reply:
Sorry about the delay in review, I was out last week)

Reply:
I agree that what @jonahgao  would allow better backwards compatibility

However, given that we don't really provide any guarantee for compatibility now anyways (I recently made this explicit in the documentation, https://github.com/apache/arrow-datafusion/blob/8f3d1ef23f93cd4303745eba76c0850b39774d07/datafusion/proto/src/lib.rs#L37-L41) I don't think we should add complexity for these functions